searvey/usgs.py

```diff
@@ -297,16 +282,20 @@ def get_usgs_station_data(
     endtime = resolve_date(endtime)
     starttime = endtime - datetime.timedelta(days=period)
     df_iv, _ = nwis.get_iv(sites=[usgs_code], start=starttime.isoformat(), end=endtime.isoformat())
-    df_iv = normalize_usgs_station_data(df=df_iv, truncate_seconds=truncate_seconds)
+    df_iv = normalize_usgs_station_data(df=df_iv)
     return df_iv
 
 
 def _get_dataset_from_query_results(
-    query_result: Tuple[pd.DataFrame, Metadata], usgs_metadata: pd.DataFrame, truncate_seconds: bool
+    query_result: Tuple[pd.DataFrame, Metadata],
+    usgs_metadata: pd.DataFrame,
 ) -> xr.Dataset:
     df_iv, _ = query_result
+    if len(df_iv) == 0:
+        return xr.Dataset()
```

Comment:
Doesn't this change the "schema" of the `xr.Dataset` that `get_usgs_data()` function  returns? 

I mean, you can use `.sel()/.isel()` to filter all the data out of a Dataset but you still maintain, data_vars, dimensions, attributes etc. `xr.Dataset()` throws all that away. I feel that this can be a bit counter-intuitive.

Reply:
The problem is that `dataretrieval` returns empty `pandas.DataFrame` object if the response is empty. They're not defining dataframe "schema" for empty ones in their code. Either we have to explicitly define the schema in our code, or we also have to return empty like this. Since we rely on `dataretrieval` to define the schema of dataframe and we just reorganize it, I think it's reasonable to have no schema if we don't receive that info from `dataretrieval`